addons/purchase/tests/test_access_rights.py

```diff
@@ -125,3 +125,17 @@ def test_double_validation(self):
         self.purchase_user.groups_id += group_purchase_manager
         order.with_user(self.purchase_user).button_approve()
         self.assertEqual(order.state, 'purchase')
+
+    def test_create_product_purchase_user(self):
+        self.env['res.config.settings'].create({'group_uom': True}).execute()
+        uom_category_unit = self.env['uom.category'].create({'name': 'test_uom_category_unit'})
+        uom_unit = self.env['uom.uom'].create({'name':'test_unit', 'category_id': uom_category_unit.id})
+        self.purchase_user.groups_id += self.env.ref('product.group_product_manager')
+        try:
+            _ = self.env['product.template'].with_user(self.purchase_user).create({
+                'name': 'Test Product UOM Default',
+                'type': 'consu',
+                'uom_id': uom_unit.id,
+            })
+        except UserError as exc:
+            self.fail(f"ValidationError: {exc}")
```

Comment:
Not sure if a try-catch is really necessary here.

Could just create a product, then `self.assertEqual(product.uom_po_id, product.uom_id)` (just to illustrate the purpose of the test, as the test would fail before it gets there).

Or if you think `self.fail` makes more sense, have it return a message explaining what the expected result should've been.

Reply:
No you cannot create a product, the error is happening at creation.